app/src/main/java/com/example/countryinfo/CountryDetails.kt

```diff
@@ -0,0 +1,30 @@
+package com.example.countryinfo
+
+class CountryDetails(
+    val name: String,
+    val capital: String,
+    val region: String,
+    val population: Double,
+    val flag: String,
+    val currencies: List<Currency>,
+    val languages: List<Language>,
+    val timezones: List<TimeZone>,
+    val callingCodes: List<CallingCode>
```

Comment:
Let's keep these as regular strings instead of wrapping into classes. Not sure if we would need to extend Language, TimeZone or CallingCode in the future. Moreover we can format them appropriately while mapping from GQL models.
```suggestion
    val languages: List<String>,
    val timezones: List<String>,
    val callingCodes: List<String>
```

Reply:
The same is applicable for `Currency`, we can format it right away, e.g. `"$symbol $name"`, so no need to keep this formatting logic in the view layer later.

Reply:
Changed.